website/docs/language/tests/index.mdx

```diff
@@ -1009,4 +1014,15 @@ Run D (parallel: true)
 Run E (parallel: true)
 ```
 
-Runs A and B execute simultaneously. Run C waits until both A and B are done before starting. Finally, runs D and E run in parallel, but only after C completes.
\ No newline at end of file
+Runs A and B execute simultaneously. Run C waits until both A and B are done before starting. Finally, runs D and E run in parallel, but only after C completes.
+
+## Skipping Cleanup
+
+You can skip the cleanup process by setting the `skip_cleanup` attribute to `true` in the `test` or `run` block. This is useful for debugging or when you want to inspect the state files after a test run. 
+ When multiple `run` blocks that modify the same state file set this attribute, only the last `run` block with `skip_cleanup` set to `true` will take effect. This means the state file will reflect the state as of that last `run` block.
+ Any resources retained in the state due to this attribute will also be saved in the state file artifacts. Refer to the [Artifacts](#handling-leftover-state-artifacts) section for more information. This means that you have to cleanup the state before running any other tests.
+
+## Handling Leftover State Artifacts
+
+During the execution of a test file, Terraform automatically cleans up any state files it generates. This cleanup involves destroying the created resources. However, if the cleanup process fails, Terraform generates a manifest file that references state files containing the remaining resources. The manifest file is located within the data directory of the terraform configuration (`.terraform/test/manifest.json`).
```

Comment:
I found this section a bit confusing, particularly about when state artifacts are created and when. I believe state artifacts are only made when there's an issue with cleanup, but "Terraform automatically cleans up any state files it generates" seems to contradict that.

It may be that there needs to be a clearer distinction between the process of destroying resources represented in the internal, in-memory state versus the state files written to the .terraform/test directory?